libdnf/base/base.cpp

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/base/base.hpp"
+
+#include "libdnf/conf/config_parser.hpp"
+#include "libdnf/conf/const.hpp"
+
+#include <fmt/format.h>
+
+#include <algorithm>
+#include <filesystem>
+#include <vector>
+
+namespace libdnf {
+
+static void load_config_from_parser(
+    Config & conf, const ConfigParser & parser, const std::string & section, Logger & log) {
+    const auto & cfg_parser_data = parser.get_data();
+    auto cfg_parser_data_iter = cfg_parser_data.find(section);
+    if (cfg_parser_data_iter != cfg_parser_data.end()) {
+        auto opt_binds = conf.opt_binds();
+        const auto & cfg_parser_main_sect = cfg_parser_data_iter->second;
+        for (const auto & opt : cfg_parser_main_sect) {
+            auto opt_binds_iter = opt_binds.find(opt.first);
+            if (opt_binds_iter != opt_binds.end()) {
+                try {
+                    opt_binds_iter->second.new_string(libdnf::Option::Priority::MAINCONFIG, opt.second);
+                } catch (const Option::Exception & ex) {
+                    auto msg = fmt::format(
+                        R"**(Config error in section "{}" key "{}": {}: {})**",
+                        section,
+                        opt.first,
+                        ex.get_description(),
+                        ex.what());
+                    log.warning(msg);
+                }
+            }
+        }
+    }
+}
+
+static void load_config_from_file_path(
+    Config & conf, const std::string & file, const std::string & section, Logger & log) {
+    log.debug(fmt::format(R"**(Start of loading section "{}" from file "{}")**", section, file));
```

Comment:
Do we really want to log `Start of loading section` or that the process was done?

Reply:
I will remove the message. But in the future we must to consider which debug messages we want to log. Debug messages can help in some situations.

Reply:
IMO start and end of loading a config file (not a section) is a good threshold for the detail on the debug logging level.